### DISCUSSION TITLE
Improve separation of concern in extensions

### DISCUSSION THREAD

**Discussion by mikael-desharnais** [eclipse-theia/theia#8150] [General]
Hi,
exploring the default packages for Theia, I discover that some features are provided by some extensions without apparent logic (except maybe historical reasons). I would like to know if pull requests regarding such topics have a chance to be accepted or if it does not belong to the immediate concerns of the project.

As a example, as it is today, the Widget that allows to quick launch commands belongs to the monaco extension and so cannot be used without the monaco extension. It seems to me that its scope is larger then that of the monaco editor and it would benefit from being externalized.

**Comment by akosyakov** [comment]
How would you propose to do it?

**Reply by mikael-desharnais** [reply]
Well to me this widget should be closer to the core package (or even be part of it).
After a quick study it seems that what links this widget to the monaco editor are the MonacoContextKeyService and other Keybindings tools which may also be extracted from the monaco extension.
I say all this without any thorough study. I would just like to know if it would be worth studying this issue and proposing a PR.

**Reply by akosyakov** [reply]
These services are implemented by Monaco, not by us. You will need to move monaco to core. I am not sure how you can extract it.

Do you want to get rid of the code editor in your app?

**Reply by mikael-desharnais** [reply]
I know It sound strange to use Theia (an IDE) to create software without an IDE but the company I work for is aiming at developping a complete suite of software that would provide mostly non IDE features. We were requested to be able to provide the application without  the IDE parts.
At the moment, I was able to create an extension that provides the minimum features that are missing in the core package and that make it crash or disfunction if it is alone (mostly the ColorRegistry & the PreferenceProvider) but my application is still missing Keybindings and Quick command widget.

**Reply by akosyakov** [reply]
For the quick palette you will need Monaco, there is no a way around. What you can do it to use DI to unbind unnecessary contributions, like:
```ts
import { injectable } from 'inversify';
import { CommandRegistry } from '@theia/core';
import { EditorCommandContribution } from '@theia/editor/lib/browser';


@injectable()
export class MyEditorCommandContribution extends EditorCommandContribution {

    async registerCommands(commands: CommandRegistry): Promise<void> {
        // do nothing
    }
}
```

and then
```ts
rebind(EditorCommandContribution).to(MyEditorCommandContribution).inSingletonScope();
```

in your frontend module
